contracts/BaseTemplatesManager.sol

```diff
@@ -109,29 +109,55 @@ abstract contract BaseTemplatesManager is Ownable, IBaseTemplatesManager {
+    /// This function can only be called by the contract owner (governance).
+    /// @param _id The template's id.
+    /// @param _newSpecification the updated specification for the template with id `_id`.
+    function updateTemplateSpecification(
+        uint256 _id,
+        string calldata _newSpecification
+    ) external override onlyOwner {
+        if (bytes(_newSpecification).length == 0) revert InvalidSpecification();
+        uint128 _version = latestVersionStorageTemplate(_id).version;
+        _updateTemplateSpecification(_id, _version, _newSpecification);
+        emit UpdateTemplateSpecification(_id, _newSpecification, _version);
+    }
+
     /// @dev Updates a template specification. The specification is a cid
     /// pointing to a structured JSON file containing data about the template.
     /// This function can only be called by the contract owner (governance).
     /// @param _id The template's id.
+    /// @param _version The version of the template we want to update.
     /// @param _newSpecification the updated specification for the template with id `_id`.
     function updateTemplateSpecification(
         uint256 _id,
+        uint128 _version,
         string calldata _newSpecification
     ) external override onlyOwner {
-        // TODO: implement version update. If the version is the latest one, we need to also update the latestVersionTemplates array in addition to the templateByIdAndVersion one
         if (bytes(_newSpecification).length == 0) revert InvalidSpecification();
+        _updateTemplateSpecification(_id, _version, _newSpecification);
+        emit UpdateTemplateSpecification(_id, _newSpecification, _version);
+    }
+
+    /// @dev Internal implementation of the specification update function. It checks if the
+    /// updated version is the latest one, and in case it is it updates the `templateIdToLatestVersionIndex`
+    /// in addition to the `templateByIdAndVersion` one.
+    /// @param _id The template's id.
+    /// @param _version The version of the template we want to update.
+    /// @param _newSpecification the updated specification for the template with id `_id`.
+    function _updateTemplateSpecification(
+        uint256 _id,
+        uint128 _version,
+        string calldata _newSpecification
+    ) internal {
         Template
             storage _latestVersionStorageTemplate = latestVersionStorageTemplate(
                 _id
             );
-        _latestVersionStorageTemplate.specification = _newSpecification;
-        templateByIdAndVersion[_id][_latestVersionStorageTemplate.version]
-            .specification = _newSpecification;
-        emit UpdateTemplateSpecification(
-            _id,
-            _newSpecification,
-            _latestVersionStorageTemplate.version
-        );
+        if (_version == _latestVersionStorageTemplate.version)
+            _latestVersionStorageTemplate.specification = _newSpecification;
```

Comment:
Why do we need the condition here? Can't we just have:

```
templateByIdAndVersion[_id][_version].specification = _newSpecification;
```

And if the `_version` is the latest, it will update the latest one (I'm missing something for sure tho).

Reply:
Yeah we actually have 2 separate entities here, a mapping that returns a template from a given id and version, and an array that ONLY contains the latest versions for each template. These are 2 separate entities and they were created in order to make it easier to handle the `view` parts of the contracts (i.e. readonly queries).

We absolutely need to update both separately.